src/resolve/test-dns-packet-extract.c

```diff
@@ -0,0 +1,245 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "resolved-dns-packet.h"
+
+#include "log.h"
+#include "tests.h"
+
+#define BIT_QR (1 << 7)
+#define BIT_AA (1 << 2)
+#define BIT_TC (1 << 1)
+#define BIT_RD (1 << 0)
+
+#define BIT_RA (1 << 7)
+#define BIT_AD (1 << 5)
+#define BIT_CD (1 << 4)
+
+/* ================================================================
+ * packet header
+ * ================================================================ */
+
+TEST(packet_header_query_basic) {
+        _cleanup_(dns_packet_unrefp) DnsPacket *packet = NULL;
+
+        dns_packet_new(&packet, DNS_PROTOCOL_DNS, 0, DNS_PACKET_SIZE_MAX);
```

Comment:
This allocates memory and can fail, so wrap in ASSERT_OK

In general we always want error checks or ENOMEM checks when functions can fail due to OOM, so please update all tests across the various PRs to do so. In test code it's ok to use asserts as above, when we expect the thing to work.